src/SwedbankPay.Sdk/SwedbankPayHttpClient.cs

```diff
@@ -0,0 +1,173 @@
+            {
+                throw new BadRequestException(e);
+            }
+            catch (TaskCanceledException te)
+            {
+                throw new ApiTimeOutException(te);
+            }
+
+            if (response.IsSuccessStatusCode)
+            {
+                var res = await response.Content.ReadAsStringAsync();
+                var res2 = JToken.Parse(res).ToString(Formatting.Indented);
+                this.logger.LogInformation(res2);
+                return res2;
+            }
+
+            return null;
+        }
+
+
+        internal async Task<TResponse> HttpGet<TResponse>(string url, Func<ProblemsContainer, Exception> onError)
+            where TResponse : new()
+        {
+            return await SendHttpRequestAndProcessHttpResponse<TResponse>(HttpMethod.Get, url, onError);
+        }
+
+
+        internal async Task<TResponse> HttpPatch
+            <TPayLoad, TResponse>(string url, Func<ProblemsContainer, Exception> onError, TPayLoad payload)
+            where TResponse : new()
+        {
+            return await SendHttpRequestAndProcessHttpResponse<TResponse>(new HttpMethod("PATCH"), url, onError, payload);
+        }
+
+
+        internal async Task<TResponse> HttpPost
+            <TPayLoad, TResponse>(string url, Func<ProblemsContainer, Exception> onError, TPayLoad payload)
+            where TResponse : new()
+        {
+            return await SendHttpRequestAndProcessHttpResponse<TResponse>(HttpMethod.Post, url, onError, payload);
+        }
+
+
+        internal async Task<T> SendHttpRequestAndProcessHttpResponse
+            <T>(HttpMethod httpMethod, string url, Func<ProblemsContainer, Exception> onError, object payload = null)
+            where T : new()
+        {
+            var requestMessage = new HttpRequestMessage(httpMethod, url);
+
+            return await SendHttpRequestAndProcessHttpResponse<T>(requestMessage, onError, payload);
+        }
+
+
+        internal async Task<T> SendHttpRequestAndProcessHttpResponse
+            <T>(HttpRequestMessage requestMessage, Func<ProblemsContainer, Exception> onError, object payload = null)
+            where T : new()
+        {
+            UpdateRequest(requestMessage, payload);
+
+            HttpResponseMessage response = await this.client.SendAsync(requestMessage, HttpCompletionOption.ResponseHeadersRead).ConfigureAwait(false);
+            
+            if (response.IsSuccessStatusCode)
+            {
+                var res = await response.Content.ReadAsStringAsync();
+                this.logger.LogInformation(res);
+                return JsonConvert.DeserializeObject<T>(res, JsonSerialization.JsonSerialization.Settings);
+            }
+
+            var responseMessage = await response.Content.ReadAsStringAsync();
+            this.logger.LogInformation(responseMessage);
+            ProblemsContainer problems;
+            if (!string.IsNullOrEmpty(responseMessage) && IsValidJson(responseMessage))
```

Comment:
What if we receive an erronous HTTP status code but with valid JSON attached? There's no guarantee the contents will be deserializable to the ProblemsContainer type and this will explode.